bundles/org.openhab.binding.wizlighting/src/main/java/org/openhab/binding/wizlighting/internal/handler/WizLightingHandler.java

```diff
@@ -0,0 +1,688 @@
+            stateMap.put(CHANNEL_DIMMING, 0);
+        } else {
+            stateMap.put(CHANNEL_DIMMING, receivedParam.dimming);
+            switch (receivedParam.getColorMode()) {
+                case RGBMode:
+                    logger.debug("Received color values - R: {} G: {} B: {} W: {} C: {} Dimming: {}", receivedParam.r,
+                            receivedParam.g, receivedParam.b, receivedParam.w, receivedParam.c, receivedParam.dimming);
+                    logger.debug("Translated to HSBValues {}", receivedParam.getHSBColor());
+                    updateState(CHANNEL_COLOR, receivedParam.getHSBColor());
+                    updateState(CHANNEL_TEMPERATURE, new PercentType(50));
+                    updateState(CHANNEL_DIMMING, new PercentType(receivedParam.dimming));
+                    updateState(CHANNEL_SWITCH_STATE, OnOffType.ON);
+                    break;
+                case CTMode:
+                    // update color temperature channel
+                    logger.debug("Received color temperature: {} ({}%)", receivedParam.temp,
+                            receivedParam.getTemperaturePercent());
+                    updateState(CHANNEL_COLOR, new HSBType(new DecimalType(0), new PercentType(0),
+                            new PercentType(receivedParam.getDimming())));
+                    updateState(CHANNEL_TEMPERATURE, receivedParam.getTemperaturePercent());
+                    updateState(CHANNEL_DIMMING, new PercentType(receivedParam.dimming));
+                    updateState(CHANNEL_SWITCH_STATE, OnOffType.ON);
+                    stateMap.put(CHANNEL_TEMPERATURE, receivedParam.getTemperaturePercent().intValue());
+                case SingleColorMode:
+                    updateState(CHANNEL_COLOR, new HSBType(new DecimalType(0), new PercentType(0),
+                            new PercentType(receivedParam.getDimming())));
+                    updateState(CHANNEL_TEMPERATURE, new PercentType(50));
+                    updateState(CHANNEL_DIMMING, new PercentType(receivedParam.dimming));
+                    updateState(CHANNEL_SWITCH_STATE, OnOffType.ON);
+            }
+        }
+
+        // update scene channel
+        if (receivedParam.sceneId > 0) {
+            logger.debug("Received scene: {} ({})", receivedParam.sceneId,
+                    WizLightingLightMode.getNameFromModeId(receivedParam.sceneId));
+            updateState(CHANNEL_LIGHT_MODE, new StringType(String.valueOf(receivedParam.sceneId)));
+            stateMap.put(CHANNEL_LIGHT_MODE, receivedParam.sceneId);
+        }
+
+        // update speed channel
+        logger.debug("Received scene speed: {}", receivedParam.speed);
+        updateState(CHANNEL_DYNAMIC_SPEED, new PercentType(receivedParam.speed));
+        stateMap.put(CHANNEL_DYNAMIC_SPEED, receivedParam.speed);
+
+        // update signal strength
+        if (receivedParam.rssi != 0) {
+            int strength = -1;
+            if (receivedParam.rssi < -90) {
```

Comment:
I would expect this to be other way around? `< -90` being 0, as being the worst reception? https://www.openhab.org/docs/developer/bindings/thing-xml.html#system-state-channel-types

Reply:
Yup, I had it backwards.  Thank you!